### PR TITLE
Connector Store Enhancements

### PR DESCRIPTION
**Pull request by dinuish94**

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.esb.project/HTMLPages/index.html [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+<link href="./libs/bootstrap_3.3.6/css/bootstrap.min.css"
+	rel="stylesheet">
+<link href="./libs/jquery.scrollbar_0.2.11/jquery.scrollbar.css"
+	rel="stylesheet">
+<link href="./css/style.css" rel="stylesheet">
+</head>
+<body>
+	<div class="row container-body">
+		<div class="section-two col-xs-12 col-sm-12 col-md-12 col-lg-12">
+			<h2>WSO2 Enterprise Store</h2>
```
Comment (dakshika) [inline plugins/org.wso2.developerstudio.eclipse.esb.project/HTMLPages/index.html]: Connector store

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/connector/store/ConnectorStore.java [hunk 2]
```diff
@@ -34,65 +35,121 @@
...
+     * This will connect to connector store using provided URL and retrieve information for available connectors in the
+     * requested page and return a Object object which can be used to fetch relevant information for a particular
+     * connector.
+     * 
+     * @param url
+     * @return Object
+     * @throws NoSuchAlgorithmException
+     * @throws KeyManagementException
+     * @throws IOException
+     * @throws HttpException
```
Comment (chanikag) [inline plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/connector/store/ConnectorStore.java]: Need to update javadocs

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/connector/store/ConnectorStore.java [hunk 3]
```diff
@@ -34,65 +35,121 @@
...
+     * information for a particular connector.
+     * 
+     * @param url
+     * @return ConnectorData
+     * @throws NoSuchAlgorithmException
+     * @throws KeyManagementException
+     * @throws IOException
+     * @throws HttpException
+     */
+    public static ConnectorData getConnectorInfo_es210(HttpClient httpclient, String url, int start, int end)
```
Comment (chanikag) [inline plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/connector/store/ConnectorStore.java]: Need to update javadocs

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/servlets/ConnectorServletUtil.java [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+            InputStream is = url.openStream();
+            File targetFile = new File(zipDestination);
+            targetFile.createNewFile();
+            FileOutputStream outputStream = new FileOutputStream(targetFile);
+            int bytesRead = -1;
+            byte[] buffer = new byte[BUFFER_SIZE];
+            while ((bytesRead = is.read(buffer)) != -1) {
+                outputStream.write(buffer, 0, bytesRead);
+            }
+            outputStream.close();
```
Comment (chanikag) [inline plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/servlets/ConnectorServletUtil.java]: Shouldn't this done in finally block? What will happen if an exception is thrown?

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/store/handlers/ConnectorStoreEarlyStartupHandler.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        connectorsContext.addServlet(ConnectorDownloadServlet.class, CONNECTOR_DOWNLOAD_SERVLET_PATH);
+        connectorsContext.addServlet(ConnectorSearchServlet.class, CONNECTOR_SEARCH_SERVLET_PATH);
+        connectorsContext.addServlet(DefaultServlet.class, "/");
+        connectorsContext.setResourceBase(webAppPath);
+
+        // Registering the handler in the jetty server
+        jettyServerHandler.getHandlerCollection().addHandler(connectorsContext);
+        try {
+            connectorsContext.start();
+        } catch (Exception e) {
```
Comment (chanikag) [inline plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/store/handlers/ConnectorStoreEarlyStartupHandler.java]: Shall we catch the specific exception ?

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/store/handlers/ConnectorStoreEarlyStartupHandler.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+        // Context path where static webpages are hosted
+        String webAppPath = "";
+        try {
+            // Get web app path from current bundle
+            webAppPath = getWebAppPath();
+        } catch (URISyntaxException uriException) {
+            log.error("Error resolving web app path", uriException);
+        } catch (IOException ioException) {
+            log.error("Error resolving web app path", ioException);
```
Comment (chanikag) [inline plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/store/handlers/ConnectorStoreEarlyStartupHandler.java]: We can catch both IOException and URISyntaxException in a single catch block?

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/store/handlers/ConnectorStoreEarlyStartupHandler.java (hunk 5 above)
Comment (dinuish94) [reply plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/store/handlers/ConnectorStoreEarlyStartupHandler.java]: start() method in org.eclipse.jetty.util.component.LifeCycle throws Exception

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/store/handlers/ConnectorStoreEarlyStartupHandler.java (hunk 6 above)
Comment (dinuish94) [reply plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/store/handlers/ConnectorStoreEarlyStartupHandler.java]: Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/1063

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/servlets/ConnectorServletUtil.java (hunk 4 above)
Comment (dinuish94) [reply plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/servlets/ConnectorServletUtil.java]: Fixed in #1063

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/connector/store/ConnectorStore.java (hunk 3 above)
Comment (dinuish94) [reply plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/connector/store/ConnectorStore.java]: Fixed in #1063

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/connector/store/ConnectorStore.java (hunk 2 above)
Comment (dinuish94) [reply plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/connector/store/ConnectorStore.java]: Fixed in #1063

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/servlets/FunctionServerConstants.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2009-2018 WSO2, Inc. (http://wso2.com)
```
Comment (mohanvive) [inline plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/servlets/FunctionServerConstants.java]: Why, above copyright year says 2009-2018?

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/ui/wizard/CloudConnectorImportWizard.java [hunk 8]
```diff
@@ -48,224 +48,227 @@
...
+        // downloadConnectorAndUpdateProjects(downloadLink);
+        // monitor.worked(1);
+        // count++;
+        // }
+        // monitor.done();
+        // return Status.OK_STATUS;
+        // }
+        // };
+        // downloadJob.schedule();
+        // Refresh project
```
Comment (mohanvive) [inline plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/ui/wizard/CloudConnectorImportWizard.java]: Why do we have commented code here? Can't we remove them?

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/ui/wizard/CloudConnectorImportWizard.java [hunk 9]
```diff
@@ -48,224 +48,227 @@
...
+            File targetFile = new File(zipDestination);
+            targetFile.createNewFile();
+            FileOutputStream outputStream = new FileOutputStream(targetFile);
+            int bytesRead = -1;
+            byte[] buffer = new byte[BUFFER_SIZE];
+            while ((bytesRead = is.read(buffer)) != -1) {
+                outputStream.write(buffer, 0, bytesRead);
+            }
+            outputStream.close();
+            is.close();
```
Comment (mohanvive) [inline plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/ui/wizard/CloudConnectorImportWizard.java]: These, IO streams should be closed within finally block.

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/ui/wizard/ImportCloudConnectorWizardPage.java [hunk 10]
```diff
@@ -247,11 +195,11 @@ public static IProject getProject(Object obj){
...
-		if ((getCloudConnectorPath() == null || getCloudConnectorPath().equals(""))) {
-			setErrorMessage("Please specify a connector path");
-			setPageComplete(false);
-			return;
-		}
+//		if ((getCloudConnectorPath() == null || getCloudConnectorPath().equals(""))) {
+//			setErrorMessage("Please specify a connector path");
+//			setPageComplete(false);
+//			return;
+//		}
```
Comment (mohanvive) [inline plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/ui/wizard/ImportCloudConnectorWizardPage.java]: Can we remove these commented code?

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/ui/wizard/CloudConnectorImportWizard.java (hunk 8 above)
Comment (dinuish94) [reply plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/ui/wizard/CloudConnectorImportWizard.java]: Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/832

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/ui/wizard/ImportCloudConnectorWizardPage.java (hunk 10 above)
Comment (dinuish94) [reply plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/ui/wizard/ImportCloudConnectorWizardPage.java]: Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/832

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/ui/wizard/CloudConnectorImportWizard.java (hunk 9 above)
Comment (dinuish94) [reply plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/ui/wizard/CloudConnectorImportWizard.java]: Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/832

File: plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/servlets/FunctionServerConstants.java (hunk 7 above)
Comment (dinuish94) [reply plugins/org.wso2.developerstudio.eclipse.esb.project/src/org/wso2/developerstudio/eclipse/esb/project/servlets/FunctionServerConstants.java]: Fixed in https://github.com/wso2/devstudio-tooling-esb/pull/1063

**Pull request merged**